src/create-matcher.js

```diff
@@ -198,6 +203,26 @@ function fillParams (
   }
 }
 
+function getParams (path: string): Array<string> {
+  const hit = regexpParamsCache[path]
+
+  if (hit) {
+    return hit
+  }
+
+  const keys = []
+  const paramNames: Array<string> = []
+
+  Regexp(path, keys)
```

Comment:
We don't have to call `Regexp(path)` here again (without cache), there's already code for getting cached keys and regexp here (https://github.com/znck/vue-router/blob/ab0a779ba6f09f5b28138c9445ce84d7263daef6/src/create-matcher.js#L163-L172). 

We should probably create a separate function that uses the code above, take a `path` as only parameter, returns a `{keys, regexp}` object, maybe name it `getKeysAndParams`.